src/main/java/net/datafaker/Unique.java

```diff
@@ -0,0 +1,60 @@
+package net.datafaker;
+
+import java.util.*;
+
+public class Unique extends AbstractProvider {
+
+    private final Map<Locale, Map<String, List<String>>> valuesByKeyAndLocale = new HashMap<>();
+
+    public Unique(Faker faker) {
+        super(faker);
+    }
+
+    public String fetchFromYaml(String key) {
+        Locale locale = faker.getLocale();
+
+        Map<String, List<String>> valuesByKey = valuesByKeyAndLocale.getOrDefault(locale, new HashMap<>());
+        List<String> values = valuesByKey.get(key);
+
+        if (values == null) {
+            values = fetchValues(key);
+        }
+
+        if (values.isEmpty()) {
+            throw new RuntimeException(String.format(
+                "All possible values have been generated for key %s under locale %s",
+                key,
+                locale
+            ));
+        }
+
+        int index = faker.random().nextInt(0, values.size() - 1);
+        String value = values.remove(index);
+
+        valuesByKey.put(key, values);
+        valuesByKeyAndLocale.put(locale, valuesByKey);
+
+        return value;
+    }
+
+    @SuppressWarnings("unchecked")
+    private List<String> fetchValues(String key) {
+        Object object = faker.fakeValuesService().fetchObject(key);
+
+        if (object == null) {
+            throw new RuntimeException(String.format(
+                "No values found for key %s",
+                key
+            ));
+        }
+
+        if (!(object instanceof List)) {
+            throw new RuntimeException(String.format(
+                "Object found for key %s is not in list format",
+                key
+            ));
+        }
+
+        return new ArrayList<>((List<String>) object);
```

Comment:
There is no check that it is a list o strings and not a list of lists....
So probably in case it is a list of lists it makes sense to throw an exception like `there is not value for such a key`

Reply:
I added a check that filters out any lists and throws the same "No values found for key" exception. Is RuntimeException okay for that or is there a more suitable exception for this case?